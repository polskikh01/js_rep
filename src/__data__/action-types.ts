export const PRODUCT = {
    REQUEST: 'PRODUCT_REQUEST',
    SUCCESS: 'PRODUCT_SUCCESS',
    FAILURE: 'PRODUCT_FAILURE',
};

export const BASKET = {
    REQUEST: 'BASKET_REQUEST',
    SUCCESS: 'BASKET_SUCCESS',
    FAILURE: 'BASKET_FAILURE',
};

export const ACCESSORIES = {
    REQUEST: 'ACCESSORIES_REQUEST',
    SUCCESS: 'ACCESSORIES_SUCCESS',
    FAILURE: 'ACCESSORIES_FAILURE',
};

export const COMPLETE = {
    REQUEST: 'COMPLETE_REQUEST',
    SUCCESS: 'COMPLETE_SUCCESS',
    FAILURE: 'COMPLETE_FAILURE',
};