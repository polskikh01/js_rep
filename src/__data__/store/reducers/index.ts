import { combineReducers } from 'redux';

import productReducer, {ProductState} from './product'

export type AppStore = {
    product: ProductState;
};

export default combineReducers<AppStore>({
    product: productReducer
})