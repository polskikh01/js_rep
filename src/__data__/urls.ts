import { getNavigations } from '@ijl/cli';

const navigations = getNavigations('js_rep');

export const baseUrl = navigations['js_rep'];

export const URLs = {
    main: {
        url: navigations['link.js_rep.main'],
    },
    product: {
        url: navigations['link.js_rep.product'],
    },
    accessories: {
        url: navigations['link.js_rep.accessories'],
    },
    complete: {
        url: navigations['link.js_rep.complete'],
    },
    contacts: {
        url: navigations['link.js_rep.contacts'],
    },
    shippingAndPayment: {
        url: navigations['link.js_rep.shippingAndPayment']
    }
}