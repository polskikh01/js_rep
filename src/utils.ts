import {
    classic,
    mini,
    ready,
    simple,
    miniK,
    magnit,
    killerM,
    glinaClassic,
    glinaMini,
    white,
    black,
    connector,
    classicK,
} from './asset/index';

export function getImgByName(name) {
    switch (name) {
        case 'simple':
            return simple;
        case 'classic':
            return classic;
        case 'mini':
            return mini;
        case 'ready':
            return ready;
        case 'miniK':
            return miniK;
        case 'magnit':
            return magnit;
        case 'killerM':
            return killerM;
        case 'glinaClassic':
            return glinaClassic;
        case 'glinaMini':
            return glinaMini;
        case 'white':
            return white;
        case 'black':
            return black;
        case 'connector':
            return connector;
        case 'classicK':
            return classicK;
    }
}
