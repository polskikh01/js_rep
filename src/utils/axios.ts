import axios from 'axios';
import { getConfigValue } from '@ijl/cli';

const baseApiUrl = getConfigValue('js_rep');
export const commonAxios = axios.create({
    baseURL: baseApiUrl,
    headers: {
        'Content-Type': 'application/json;charset=utf-8',
    },
});
