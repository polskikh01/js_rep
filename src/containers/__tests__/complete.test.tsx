import React from 'react';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import { describe, it, expect, beforeEach } from '@jest/globals';
import { Provider } from 'react-redux';
import mockAdapter from 'axios-mock-adapter';

import { commonAxios } from '../../utils/axios';
import product_mocks from '../../../stubs/api/mocks/complete/success.json';

import { store } from '../../__data__/store';

import Complete from '../complete/complete';

const multipleRequest = async (mock, responses) => {
    await act(async () => {
        await mock.onAny().reply((config) => {
            const [method, url, params, ...response] = responses.shift();
            if (config.url.includes(url)) {
                return response;
            }
        });
    });
};

describe('Тестирование всего приложения', () => {
    let mockAxios;
    beforeEach(() => {
        mockAxios = new mockAdapter(commonAxios);
    });
    it('Тестируем рендер Complete', async () => {
        expect.assertions(2);
        const component = mount(
            <Provider store={store}>
                <Complete />
            </Provider>
        );

        expect(component).toMatchSnapshot();

        const response = [
            [
                'GET',
                `/getComplete`,
                {},
                200,
                {
                    ...product_mocks,
                },
            ],
        ];

        await multipleRequest(mockAxios, response);
        component.update();

        expect(component).toMatchSnapshot();
    });
});
