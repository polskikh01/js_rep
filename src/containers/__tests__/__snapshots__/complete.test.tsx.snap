// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование всего приложения Тестируем рендер Complete 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Complete)>
    <Complete
      addProduct={[Function]}
      getComplete={[Function]}
      loading={true}
      productItems={Array []}
    >
      <p>
        loading...
      </p>
    </Complete>
  </Connect(Complete)>
</Provider>
`;

exports[`Тестирование всего приложения Тестируем рендер Complete 2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Complete)>
    <Complete
      addProduct={[Function]}
      getComplete={[Function]}
      loading={false}
      productItems={
        Array [
          Object {
            "photo": "white",
            "price": "750",
            "title": "js_rep.complete.hoseW",
          },
          Object {
            "photo": "black",
            "price": "750",
            "title": "js_rep.complete.hoseB",
          },
          Object {
            "photo": "magnit",
            "price": "80",
            "title": "js_rep.complete.magnet",
          },
          Object {
            "photo": "connector",
            "price": "600",
            "title": "js_rep.complete.connector",
          },
          Object {
            "photo": "classicK",
            "price": "1200",
            "title": "js_rep.complete.flaskC",
          },
          Object {
            "photo": "miniK",
            "price": "1000",
            "title": "js_rep.complete.flaskM",
          },
        ]
      }
    >
      <div
        id="wrapper"
      >
        <div
          className="focusWrapper"
          id="focus-wrapper"
        >
          <div
            className="page"
          >
            <Head>
              <header
                className="mHeader"
              >
                <div
                  className="menuHeader"
                >
                  <div
                    className="menu"
                  >
                    <Link
                      className="logo"
                      to="/main"
                    >
                      <a>
                        mock link
                      </a>
                    </Link>
                    <nav
                      className="navP"
                    >
                      <Nav
                        link="/product"
                        title="Продукты"
                      >
                        <Link
                          className="navButt"
                          to="/product"
                        >
                          <a>
                            mock link
                          </a>
                        </Link>
                      </Nav>
                      <Nav
                        link="/shippingAndPayment"
                        title="Доставка и оплата"
                      >
                        <Link
                          className="navButt"
                          to="/shippingAndPayment"
                        >
                          <a>
                            mock link
                          </a>
                        </Link>
                      </Nav>
                      <Nav
                        link="/contacts"
                        title="Контакты"
                      >
                        <Link
                          className="navButt"
                          to="/contacts"
                        >
                          <a>
                            mock link
                          </a>
                        </Link>
                      </Nav>
                    </nav>
                    <div
                      className="basket"
                    >
                      <Link
                        to="/basket"
                      >
                        <a>
                          mock link
                        </a>
                      </Link>
                    </div>
                  </div>
                </div>
              </header>
            </Head>
            <div
              className="wrap"
            >
              <span
                key="0"
              >
                <LotAlt
                  click={[Function]}
                  photo="FILE_MOCK"
                  price="750"
                  text="Купить"
                  title="Шланг белый"
                >
                  <div
                    className="product"
                  >
                    <span
                      className="productTitle"
                    >
                      Шланг белый
                    </span>
                    <p />
                    <span>
                      750
                      ₽
                    </span>
                    <button
                      onClick={[Function]}
                    >
                      <span
                        className="productBuy"
                      >
                        Купить
                      </span>
                    </button>
                    <img
                      src="FILE_MOCK"
                    />
                  </div>
                </LotAlt>
              </span>
              <span
                key="1"
              >
                <LotAlt
                  click={[Function]}
                  photo="FILE_MOCK"
                  price="750"
                  text="Купить"
                  title="Шланг черный"
                >
                  <div
                    className="product"
                  >
                    <span
                      className="productTitle"
                    >
                      Шланг черный
                    </span>
                    <p />
                    <span>
                      750
                      ₽
                    </span>
                    <button
                      onClick={[Function]}
                    >
                      <span
                        className="productBuy"
                      >
                        Купить
                      </span>
                    </button>
                    <img
                      src="FILE_MOCK"
                    />
                  </div>
                </LotAlt>
              </span>
              <span
                key="2"
              >
                <LotAlt
                  click={[Function]}
                  photo="FILE_MOCK"
                  price="80"
                  text="Купить"
                  title="Неодовый магнит"
                >
                  <div
                    className="product"
                  >
                    <span
                      className="productTitle"
                    >
                      Неодовый магнит
                    </span>
                    <p />
                    <span>
                      80
                      ₽
                    </span>
                    <button
                      onClick={[Function]}
                    >
                      <span
                        className="productBuy"
                      >
                        Купить
                      </span>
                    </button>
                    <img
                      src="FILE_MOCK"
                    />
                  </div>
                </LotAlt>
              </span>
              <span
                key="3"
              >
                <LotAlt
                  click={[Function]}
                  photo="FILE_MOCK"
                  price="600"
                  text="Купить"
                  title="Коннектор"
                >
                  <div
                    className="product"
                  >
                    <span
                      className="productTitle"
                    >
                      Коннектор
                    </span>
                    <p />
                    <span>
                      600
                      ₽
                    </span>
                    <button
                      onClick={[Function]}
                    >
                      <span
                        className="productBuy"
                      >
                        Купить
                      </span>
                    </button>
                    <img
                      src="FILE_MOCK"
                    />
                  </div>
                </LotAlt>
              </span>
              <span
                key="4"
              >
                <LotAlt
                  click={[Function]}
                  photo="FILE_MOCK"
                  price="1200"
                  text="Купить"
                  title="Колба Classic"
                >
                  <div
                    className="product"
                  >
                    <span
                      className="productTitle"
                    >
                      Колба Classic
                    </span>
                    <p />
                    <span>
                      1200
                      ₽
                    </span>
                    <button
                      onClick={[Function]}
                    >
                      <span
                        className="productBuy"
                      >
                        Купить
                      </span>
                    </button>
                    <img
                      src="FILE_MOCK"
                    />
                  </div>
                </LotAlt>
              </span>
              <span
                key="5"
              >
                <LotAlt
                  click={[Function]}
                  photo="FILE_MOCK"
                  price="1000"
                  text="Купить"
                  title="Колба Mini"
                >
                  <div
                    className="product"
                  >
                    <span
                      className="productTitle"
                    >
                      Колба Mini
                    </span>
                    <p />
                    <span>
                      1000
                      ₽
                    </span>
                    <button
                      onClick={[Function]}
                    >
                      <span
                        className="productBuy"
                      >
                        Купить
                      </span>
                    </button>
                    <img
                      src="FILE_MOCK"
                    />
                  </div>
                </LotAlt>
              </span>
            </div>
            <Footer>
              <footer
                className="footerM"
              >
                <div
                  className="footerContainer"
                >
                  <span>
                    Calabash
                  </span>
                  <h6>
                    ©2020-2021
                  </h6>
                </div>
                <div
                  className="footerContainer"
                >
                  <span
                    className="title"
                  >
                    Продукты
                  </span>
                  <Link
                    to="/product"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                  <Link
                    to="/accessories"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                  <Link
                    to="/complete"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                </div>
                <div
                  className="footerContainer"
                >
                  <span
                    className="title"
                  >
                    Аккаунт
                  </span>
                  <span
                    className="fCT"
                  >
                    Личный кабинет
                  </span>
                </div>
                <div
                  className="footerContainer"
                >
                  <span
                    className="title"
                  >
                    Помощь
                  </span>
                  <Link
                    to="/shippingAndPayment"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                  <Link
                    to="/contacts"
                  >
                    <a>
                      mock link
                    </a>
                  </Link>
                </div>
              </footer>
            </Footer>
          </div>
        </div>
      </div>
    </Complete>
  </Connect(Complete)>
</Provider>
`;
